test/cpp/end2end/xds/xds_csds_end2end_test.cc

```diff
@@ -284,6 +285,51 @@ class ClientStatusDiscoveryServiceTest : public XdsEnd2endTest {
     return response;
   }
 
+  std::vector<::testing::Matcher<
+      envoy::service::status::v3::ClientConfig_GenericXdsConfig>>
+  PrepareClientConfigMatchers(bool enable_rds_testing,
```

Comment:
Please don't refactor test data out of the individual tests, since it makes it harder to understand what is happening in the test without reading code outside of the test.  If we have to duplicate this block of code in two tests, that's preferable.

Reply:
I don't see why we need this helper in the new test.  I think we can construct the list of matchers inside the new test, just like we're doing with the existing test.